### PR TITLE
[#68] EKG & Prometheus loaded from plugin

### PR DESCRIPTION
**Pull request by CodiePP**
EKG (and Prometheus) loaded from plugin.
listens again on http://localhost:12799/metrics

### CODE REVIEW

File: src/exec/Logging.hs [hunk 1]
```diff
@@ -32,7 +35,15 @@ withLogging mLoggerConfig name k = do
   -- iohk-monitoring uses some MVar for configuration, which corresponds to
   -- the "Representation" which we call config.
   loggerConfig' <- Monitoring.setupFromRepresentation loggerConfig
-  Monitoring.withTrace loggerConfig' name k
+  bracket
+    (do (tr0, sb0) <- Monitoring.setupTrace_ loggerConfig' name
+        -- load plugin for EKG and Prometheus backend
+        Cardano.BM.Backend.EKGView.plugin loggerConfig' tr0 sb0
```
Comment (avieth) [inline src/exec/Logging.hs:41]: Does this load EKG always, or does it only do it if the config says it should?

**Review by avieth — APPROVED** [review]

File: src/exec/Logging.hs (hunk 1 above)
Comment (CodiePP) [reply src/exec/Logging.hs:41]: it loads it permanently, but only activates it if the listening port definition is present in the configuration.

Comment (CodiePP) [reply src/exec/Logging.hs:41]: same for the Prometheus interface: it will only be activated if the port is defined in the config.

Comment (ArturWieczorek) [reply src/exec/Logging.hs:41]: I checkout this PR branch and run `cabal new-update`   `cabal new-build all` and got this error:      

> Building` library for dns-4.0.1..
> 
> [ 1 of 10] Compiling Network.DNS.Decode ( Network/DNS/Decode.hs, dist/build/Network/DNS/Decode.o )
> [ 2 of 10] Compiling Network.DNS.Encode ( Network/DNS/Encode.hs, dist/build/Network/DNS/Encode.o )
> [ 3 of 10] Compiling Network.DNS.IO   ( Network/DNS/IO.hs, dist/build/Network/DNS/IO.o )
> Network/DNS/IO.hs:163:55: error:
>     • Couldn't match type 'IPv4'
>                      with 'iproute-1.7.7:Data.IP.Addr.IPv4'
>       NB: 'iproute-1.7.7:Data.IP.Addr.IPv4'
>             is defined in 'Data.IP.Addr' in package 'iproute-1.7.7'
>           'IPv4' is defined in 'Data.IP.Addr' in package 'iproute-1.7.7'
>       Expected type: [iproute-1.7.7:Data.IP.Addr.IPv4]
>         Actual type: [IPv4]
>     • In the second argument of '(<$>)', namely 'ips'
>       In the expression: ResourceRecord dom A classIN 300 . RD_A <$> ips
>       In an equation for 'as':
>           as = ResourceRecord dom A classIN 300 . RD_A <$> ips
>     |
> 163 |     as  = ResourceRecord dom A classIN 300 . RD_A <$> ips
>     |                                                       ^^^
> 
> Network/DNS/IO.hs:176:61: error:
>     • Couldn't match type 'IPv6'
>                      with 'iproute-1.7.7:Data.IP.Addr.IPv6'
>       NB: 'iproute-1.7.7:Data.IP.Addr.IPv6'
>             is defined in 'Data.IP.Addr' in package 'iproute-1.7.7'
>           'IPv6' is defined in 'Data.IP.Addr' in package 'iproute-1.7.7'
>       Expected type: [iproute-1.7.7:Data.IP.Addr.IPv6]
>         Actual type: [IPv6]
>     • In the second argument of '(<$>)', namely 'ips'
>       In the expression:
>         ResourceRecord dom AAAA classIN 300 . RD_AAAA <$> ips
>       In an equation for 'as':
>           as = ResourceRecord dom AAAA classIN 300 . RD_AAAA <$> ips
>     |
> 176 |     as  = ResourceRecord dom AAAA classIN 300 . RD_AAAA <$> ips
>     |                                                             ^^^
> cabal: Failed to build dns-4.0.1 (which is required by test:test-network from
> ouroboros-network-0.1.0.0, exe:demo-chain-sync from ouroboros-network-0.1.0.0
> and others). See the build log above for details.

Comment (CodiePP) [reply src/exec/Logging.hs:41]: this has probably something to do with your local setup. sorry.

**Comment by CodiePP** [comment]
bors try

let's ask `bors` if it can build it.

**Pull request merged**